### PR TITLE
[wasm][debugger] Small improvements to fail gracefully

### LINKED ISSUE #41744: [wasm][debugger] Fails to show locals, or object properties
The debugger fails to show any locals, or some object properties in these cases:

- Unhandled/unsupported types (like `out`)
- Multicast delegates, like when looking at code in aspnetcore's `EventCallback.cs`
- `ref` valuetypes
- when locals, or params could not be described for some reason (like being in on the wrong frame!)

### LINKED ISSUE #41818: [wasm][debugger] Cannot inspect locals in trace with calls made through reflection
If a stacktrace contains calls made through reflection (`methodInfo.Invoke(..)`), then locals on any frame before that call don't show up correctly, or in some cases cause a crash.

### PR DESCRIPTION
**Pull request by radical**
- Try to fail gracefully when we come across some variable/value that we can't describe (yet!)
- Eg. missing because of invalid param/local id, or value description
failed because of some as yet unsupported type

Fixes https://github.com/dotnet/runtime/issues/41744
Fixes https://github.com/dotnet/runtime/issues/41818

### CODE REVIEW

**Comment by radical** [comment]
This is based on #41562 .

**Comment by radical** [comment]
You can see just the changes for this PR: https://github.com/radical/runtime/compare/wasm-boxing...radical:wasm-misc-fixes

File: src/mono/wasm/runtime/library_mono.js [hunk 1]
```diff
@@ -362,43 +362,97 @@ var MonoSupportLib = {
...
+			let backing_fields = {};
 
-				names_found [p.name] = p.name;
-				final_var_list.push (p);
-			}
+			// Split props into the 3 groups - backing_fields, getters, and all_fields_except_backing_fields
+			props.forEach(p => {
+				if (p.name === undefined) {
+					console.debug(`Bug: Found a member with no name. Skipping it. p: ${JSON.stringify(p)}`);
+					return;
```
Comment (thaystg) [inline src/mono/wasm/runtime/library_mono.js:405]: Should we return or we can just continue the forEach for the next member?

Comment (radical) [reply src/mono/wasm/runtime/library_mono.js:405]: It's a lambda, so return here will effectively mean - continue.

**Review by thaystg — APPROVED** [review]

**Comment by radical** [comment]
Opened an issue corresponding to this - https://github.com/dotnet/runtime/issues/41744

**Review by thaystg — APPROVED** [review]

**Comment by radical** [comment]
Rebased again on https://github.com/dotnet/runtime/pull/41562 . 
Changes specific to this PR - https://github.com/radical/runtime/compare/wasm-boxing...radical:wasm-misc-fixes

**Comment by radical** [comment]
This is rebased on master now, so can be reviewed independently.

**Review by lewing — COMMENTED** [review]
the frame_id logic is getting a little odd now, it appears to start at 1 in the mono response but gets reduced to 0 in the debugger response?  have you verified that stack inspection works properly when selecting different frames (we do test for this now right?)

**Comment by radical** [comment]
> the frame_id logic is getting a little odd now, it appears to start at 1 in the mono response but gets reduced to 0 in the debugger response? have you verified that stack inspection works properly when selecting different frames (we do test for this now right?)

MonoProxy is the one doing `-1`, the IDE side will always see that final number only. But I agree it is a little.. odd. I can fix it up in debugger.c itself.
Yeah, I did test for this. And we do have some tests for it.

**Comment by radical** [comment]
> > the frame_id logic is getting a little odd now, it appears to start at 1 in the mono response but gets reduced to 0 in the debugger response? have you verified that stack inspection works properly when selecting different frames (we do test for this now right?)
> 
> MonoProxy is the one doing `-1`, the IDE side will always see that final number only. But I agree it is a little.. odd. I can fix it up in debugger.c itself.
> Yeah, I did test for this. And we do have some tests for it.

And that commit adds a test specifically for checking a previous frame. In the case of the test, we skip some frames too, so discrepancies in indexing should show up.

**Review by lewing — APPROVED** [review]

**Pull request merged**